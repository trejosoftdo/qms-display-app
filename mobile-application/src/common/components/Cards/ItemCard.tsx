import * as React from 'react';
import { StyleSheet } from 'react-native';
import { Avatar, Card, Text } from 'react-native-paper';
import AppLightTheme from '../../theme';

/**
 * ItemCardProps defines the props for the Cards Component.
 *
 * @property icon - card icon
 * @property title - card title
 * @property onPress - on press card handler
 */
interface ItemCardProps {
  icon: string;
  title: string
  onPress: () => void;
}

/**
 * A component for an item card
 *
 * @param {ItemCardProps} props - The props for the Item Card component.
 */
const ItemCard: React.FC<ItemCardProps> = (props: ItemCardProps) => (
  <Card
    mode="outlined"
    style={styles.container}
    onPress={props.onPress}
  >
    <Card.Content style={styles.content}>
      <Avatar.Icon style={styles.icon} size={64} icon={props.icon} />
      <Text variant="titleMedium" style={styles.title}>{props.title}</Text>
    </Card.Content>
  </Card>
);

const styles = StyleSheet.create({
  container: {
    backgroundColor: AppLightTheme.colors.background,
    marginRight: 2,
    cursor: 'pointer',
    width: '45%',
    maxWidth: 150,
    paddingBottom: 16,
    borderColor: AppLightTheme.colors.outline
  },
  icon: {
    backgroundColor: AppLightTheme.colors.primary,
  },
  content: {
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center'
  },
  title: {
    color: AppLightTheme.colors.secondary,
    paddingTop: 10,
    textAlign: 'center',
  },
});

export default ItemCard;