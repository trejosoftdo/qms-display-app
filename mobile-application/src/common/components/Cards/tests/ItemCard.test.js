import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import ItemCard from '../ItemCard';


describe('ItemCard component', () => {
  const props = {
    icon: 'lock',
    title: 'Test title',
    onPress: jest.fn(),
  };

  it('renders the component with the provided props as expected', () => {
    expect(renderShallow(<ItemCard {...props} />)).toMatchSnapshot();
  });
});
