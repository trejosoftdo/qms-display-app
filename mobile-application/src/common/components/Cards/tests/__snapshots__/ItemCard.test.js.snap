// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemCard component renders the component with the provided props as expected 1`] = `
<Card
  mode="outlined"
  onPress={[MockFunction]}
  style={
    {
      "backgroundColor": "#ffffff",
      "borderColor": "#d4d4d4",
      "cursor": "pointer",
      "marginRight": 2,
      "maxWidth": 150,
      "paddingBottom": 16,
      "width": "45%",
    }
  }
>
  <Card.Content
    style={
      {
        "alignItems": "center",
        "display": "flex",
        "justifyContent": "center",
      }
    }
  >
    <Avatar.Icon
      icon="lock"
      size={64}
      style={
        {
          "backgroundColor": "#2929cc",
        }
      }
    />
    <ForwardRef(Text)
      style={
        {
          "color": "#3c3744",
          "paddingTop": 10,
          "textAlign": "center",
        }
      }
      variant="titleMedium"
    >
      Test title
    </ForwardRef(Text)>
  </Card.Content>
</Card>
`;
